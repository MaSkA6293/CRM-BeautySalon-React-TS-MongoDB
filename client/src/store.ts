import { createStore, compose, applyMiddleware } from "redux";
import rootReducer from "./reducers/rootReducer";
//import thunk from "redux-thunk";

import createSagaMiddleware from "redux-saga";
import rootSaga from "./sagas";

const sagaMiddleware = createSagaMiddleware();
const composeEnhancers =
  process.env.NODE_ENV !== "production" &&
    typeof window === "object" &&
    (window as any).__REDUX_DEVTOOLS_EXTENSION_COMPOSE__
    ? (window as any).__REDUX_DEVTOOLS_EXTENSION_COMPOSE__({})
    : compose;

const configureStore = () =>
  createStore(
    rootReducer,
  //  composeEnhancers(applyMiddleware(thunk, sagaMiddleware))
    composeEnhancers(applyMiddleware( sagaMiddleware))
  );

const store = configureStore();
sagaMiddleware.run(rootSaga);
export default store;
