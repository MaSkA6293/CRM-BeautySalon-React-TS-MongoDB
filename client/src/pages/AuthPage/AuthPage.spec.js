import React from "react";
import { AuthPage } from "./index";
import { Provider } from "react-redux";
import store from "../../store";

jest.mock("react-router-dom", () => ({
    useHistory: () => ({
        push: jest.fn(),
    }),
}));

const setUp = (props) =>
    render(
        <Provider store={store}>
            <AuthPage {...props} />
        </Provider>,
    );

describe("all tests AuthPage", () => {
    let component;
    describe("AuthPage component", () => {
        it("should render AuthPage variant:true", () => {
            const authPageProps = {
                signIn: false,
                signUp: false,
                variant: true,
            };
            component = setUp(authPageProps);
            expect(component).toMatchSnapshot();
        });
        it("should render AuthPage signIn:true", () => {
            const authPageProps = {
                signIn: true,
                signUp: false,
                variant: false,
            };
            component = setUp(authPageProps);
            expect(component).toMatchSnapshot();
        });
        it("should render AuthPage signUp:true", () => {
            const authPageProps = {
                signIn: false,
                signUp: true,
                variant: false,
            };
            component = setUp(authPageProps);
            expect(component).toMatchSnapshot();
        });
    });
});
