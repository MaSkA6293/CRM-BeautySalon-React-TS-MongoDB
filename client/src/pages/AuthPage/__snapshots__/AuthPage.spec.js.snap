// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`all tests AuthPage AuthPage component should render AuthPage signIn:true 1`] = `
<div
  class="contaner"
>
  <div
    class="auth"
  >
    <h1
      class="auth__title"
    >
      Авторизация
    </h1>
    <form
      action="#"
      class="auth__form form"
    >
      <div
        class="MuiFormControl-root MuiTextField-root form__item MuiFormControl-fullWidth"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
          data-shrink="false"
        >
          Email
        </label>
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
        >
          <input
            aria-invalid="false"
            autocomplete="false"
            class="MuiInputBase-input MuiInput-input"
            name="email"
            type="email"
            value=""
          />
        </div>
      </div>
      <p
        class="MuiFormHelperText-root"
        id="component-error-text"
      />
      <div
        class="MuiFormControl-root MuiTextField-root form__item MuiFormControl-fullWidth"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
          data-shrink="false"
        >
          Пароль
        </label>
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
        >
          <input
            aria-invalid="false"
            autocomplete="false"
            class="MuiInputBase-input MuiInput-input"
            name="password"
            type="password"
            value=""
          />
        </div>
      </div>
      <p
        class="MuiFormHelperText-root"
        id="component-error-text"
      />
      <div
        class="form__submit"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-fullWidth"
          tabindex="0"
          type="submit"
        >
          <span
            class="MuiButton-label"
          >
            Войти
          </span>
        </button>
      </div>
      <div
        class="form__submit"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-fullWidth"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Назад
          </span>
        </button>
      </div>
    </form>
  </div>
</div>
`;

exports[`all tests AuthPage AuthPage component should render AuthPage signUp:true 1`] = `
<div
  class="contaner"
>
  <div
    class="auth"
  >
    <h1
      class="auth__title"
    >
      Авторизация
    </h1>
    <form
      action="#"
      class="auth__form form"
    >
      <div
        class="MuiFormControl-root MuiTextField-root form__item MuiFormControl-fullWidth"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
          data-shrink="false"
        >
          Email
        </label>
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
        >
          <input
            aria-invalid="false"
            autocomplete="false"
            class="MuiInputBase-input MuiInput-input"
            name="email"
            type="email"
            value=""
          />
        </div>
      </div>
      <p
        class="MuiFormHelperText-root"
        id="component-error-text"
      />
      <div
        class="MuiFormControl-root MuiTextField-root form__item MuiFormControl-fullWidth"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
          data-shrink="false"
        >
          Пароль
        </label>
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
        >
          <input
            aria-invalid="false"
            autocomplete="false"
            class="MuiInputBase-input MuiInput-input"
            name="password"
            type="password"
            value=""
          />
        </div>
      </div>
      <p
        class="MuiFormHelperText-root"
        id="component-error-text"
      />
      <div
        class="MuiFormControl-root MuiTextField-root form__item MuiFormControl-fullWidth"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
          data-shrink="false"
        >
          Подтвердить пароль
        </label>
        <div
          class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
        >
          <input
            aria-invalid="false"
            autocomplete="false"
            class="MuiInputBase-input MuiInput-input"
            name="confirmPassword"
            type="password"
            value=""
          />
        </div>
      </div>
      <p
        class="MuiFormHelperText-root"
        id="component-error-text"
      />
      <div
        class="form__submit"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained form__signUp MuiButton-containedPrimary MuiButton-fullWidth"
          tabindex="0"
          type="submit"
        >
          <span
            class="MuiButton-label"
          >
            Регистрация
          </span>
        </button>
      </div>
      <div
        class="form__submit"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-fullWidth"
          tabindex="0"
          type="submit"
        >
          <span
            class="MuiButton-label"
          >
            Назад
          </span>
        </button>
      </div>
    </form>
  </div>
</div>
`;

exports[`all tests AuthPage AuthPage component should render AuthPage variant:true 1`] = `
<div
  class="contaner"
>
  <div
    class="auth"
  >
    <h1
      class="auth__title"
    >
      Авторизация
    </h1>
    <div
      class="auth__variant"
    >
      <div
        class="auth__signIn"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
          tabindex="0"
          type="submit"
        >
          <span
            class="MuiButton-label"
          >
            Войти
          </span>
        </button>
      </div>
      <div
        class="auth__signUp"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained"
          tabindex="0"
          type="submit"
        >
          <span
            class="MuiButton-label"
          >
            Регистрация
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
